langchain4j-vearch/src/main/java/dev/langchain4j/store/embedding/vearch/SpacePropertyParam.java

```diff
@@ -15,105 +12,159 @@
  *
  * @see CreateSpaceRequest
  */
-@Getter
-@Setter
 @JsonIgnoreProperties(ignoreUnknown = true)
 @JsonInclude(NON_NULL)
 @JsonNaming(SnakeCaseStrategy.class)
 public abstract class SpacePropertyParam {
 
     protected SpacePropertyType type;
+    /**
+     * Whether to create a index or not.
+     */
+    protected Boolean index;
```

Comment:
hey @Martin7-1, could you please elaborate where this change comes from? Looks it was refactored to move the field to common class from children. 
Might this be considered a breaking change since class was defined as public? thanks!

Reply:
> Looks it was refactored to move the field to common class from children.

You are right, the `index` field just move from children to parent class.

> Might this be considered a breaking change since class was defined as public? thanks!

I think it's not a breaking change, because children's constructors are all the same parameters.

Reply:
theoretically it is possible to create own implementation that will break with new changes, example:

```
public class ExtendedSpacePropertyParam extends SpacePropertyParam {

    ExtendedSpacePropertyParam(SpacePropertyType type) {
        super(type);
    }
   // some logic
}
```
where super will no longer work. 
But it requires class created in specific package. Since SpacePropertyParam has package bound constructor, it might not be a big issue.

WDYT? @langchain4j

Reply:
I think users will not extend this class...And as you said, the class has package private constructor and users can only construct in using `Builder`.

Reply:
@Wadimz I trust @Martin7-1's judgement, he is the owner of this module